### PR TITLE
Export pattern accessibility documentation

### PR DESCRIPTION
**Pull request by chloepoulterdesign**
Closes #

There isn't an issue for this, however I am acting on the output of Fang Shih's audit of pattern accessibility. You can see her findings [here](https://ibm.box.com/s/m7bh9pul6p57fgedjse5193rxyn66ujp) (w3 log-in required).

#### Changelog

**New**

- Section entitled "Accessibility"
- Sub-sections called "Trigger", "Focus" and "Feedback"
- Images and text to explain the focus orders of triggering a Export modal, within an Export modal, and within the optional passive modal for feedback
- Anchor link to "Accessibility" section

### CODE REVIEW

File: src/pages/community/patterns/export-pattern/index.mdx [hunk 1]
```diff
@@ -150,3 +151,70 @@ A notification can be used to alert the user that an export has been successful.
...
+2. On `tab`, progress through any further fields that accept user input.
+3. On `tab`, apply focus to "Export".
+4. On `tab`, apply focus to "Cancel".
+
+#### If a sensible default value is selected or input for the user (the user is not required to complete any fields in order to export):
+
+1. On opening the modal, set the focus to the "Export" button.
+2. On `tab`, apply focus to "Cancel".
+3. On `tab`, apply focus to the first field that accepts user input. Follow
+   accessibiltiy guidance for the interaction type of this field (i.e.
```
Comment (emyarod) [inline src/pages/community/patterns/export-pattern/index.mdx:203]:
```suggestion
   accessibility guidance for the interaction type of this field (i.e.
```

**Review by jeanservaas — CHANGES_REQUESTED** [review]
Hey @chloepoulterdesign 

Thanks for all this work. I'm right in the middle of some production guidelines for usage images now and there are a few resources (even though they're not done) that I'd like to pass you to streamline a couple of images with our system.

For full UI images like this, I'm going to pass you my template file, I've symbolized the browser backgrounds we use and you can also add the file to your Sketch templates so you can grab the aspect ratio that you need

**Current:**
image

**Sketch templates:**

image

That way everything's using aspect ratios, the same background browser asset and the same margins (you can choose whether you want a narrower or a taller aspect ratio.

There's nothing technically wrong with these images... but I was confused by what you are trying to tell the user... Do you really need that decontextualized download button below the the data table? Aren't you just talking about what happens where there is not download / export location? It's especially confusing because one is in context (export) and one is not.

image

There's something going on with the scale, these two images look to be the same scale yet the filename in the top input field is much smaller than the text in the error field below — the bottom image looks to have the correct scale... so I would make them match. Also, the error field would have a white interior (not gray 10). And the button group on the modal (this is everywhere where you have a modal) has a one pixel Gray 20 ($ui-03) stroke between them for contrast accessibility 

image

Again, use the full ui template here:

image

Not sure why some of your images are getting a little blurry, but the one above this one is not:

image

I'd use a 2:1 aspect ratio here or go even shorter with multiples of 8, this is a lot of margin. I get trying to make the image sizes match... that's a good thing to do when the margin is reasonable, but here I think you have a lot of opportunities to employ one more smaller height in the mix — actually, I just noticed, the image size you're using for the export images above is even smaller than these two:

image

image

Please use the symbols I will provide you for Anatomy callouts so they match the other images in our system. Also please break these images up; i.e. the first one can be it's own image with the H5 Input required (it does need a caption within the image); the second image can the Sensible defaults provided as the H5 header and the two images can be on one canvas separated by a g20 rule.


image

When there is a sequence on one canvas, like above, please separate these images by a G20 rule with even margins around them.

image


Also, if you do happen to have to have captions in images please use Plex sans regular (not mono), G100 for captions within images. Just like this:

image

I'm going to clean up my file a little bit and I will add Sketch resources here at the end of the day.

**Comment by chloepoulterdesign** [comment]
Many many many thanks to @jeanservaas for her in-depth guidance and help.
For general changes to be made to the images in the pattern documentation, I've raised #2011 .

After lengthy talks with Jean, I've now removed the imagery in the accessibility section; there wasn't a single image that worked as discussed, but I think the section works as-is.

**Review by jeanservaas — APPROVED** [review]
@chloepoulterdesign so sorry, I misunderstood when I first reviewed this pattern, I didn't get that you just wanted to add this accessibility information to an already existing pattern! I was treating it as a net new. So yes, don't expect you to go and fix those images! Thanks Chloe.

File: src/pages/community/patterns/export-pattern/index.mdx [hunk 2]
```diff
@@ -150,3 +151,47 @@ A notification can be used to alert the user that an export has been successful.
...
 </Row>
+
+## Accessibility
+
+Adhere to
+[Carbon's modal guidelines](https://www.carbondesignsystem.com/components/modal/accessibility).
+
+### Trigger
+
+Most often, the export modal is triggered by clicking an "export" button.
```
Comment (jillianhowarth) [inline src/pages/community/patterns/export-pattern/index.mdx:162]:
```suggestion
Most often, the export modal is triggered by clicking an Export button.
```

File: src/pages/community/patterns/export-pattern/index.mdx [hunk 3]
```diff
@@ -150,3 +151,47 @@ A notification can be used to alert the user that an export has been successful.
...
+
+1. The user navigates to the Export button using the tab key. If Export is
```
Comment (jillianhowarth) [inline src/pages/community/patterns/export-pattern/index.mdx:164]:
```suggestion
1. The user navigates to the Export button using the `Tab` key. If Export is
```

File: src/pages/community/patterns/export-pattern/index.mdx [hunk 4]
```diff
@@ -150,3 +151,47 @@ A notification can be used to alert the user that an export has been successful.
...
+   launched from an overflow menu, the user may navigate to Export within the
+   overflow menu using the up/down arrow keys.
```
Comment (jillianhowarth) [inline src/pages/community/patterns/export-pattern/index.mdx:166]:
```suggestion
   overflow menu using the `Up` and `Down` arrow keys.
```

File: src/pages/community/patterns/export-pattern/index.mdx [hunk 5]
```diff
@@ -150,3 +151,47 @@ A notification can be used to alert the user that an export has been successful.
...
+2. The focus state is applied to the button/menu item.
+3. The user clicks "enter" to launch the Export modal.
```
Comment (jillianhowarth) [inline src/pages/community/patterns/export-pattern/index.mdx:168]:
```suggestion
3. The user presses `Enter` to launch the Export modal.
```

File: src/pages/community/patterns/export-pattern/index.mdx [hunk 6]
```diff
@@ -150,3 +151,47 @@ A notification can be used to alert the user that an export has been successful.
...
+### Focus
+
+When the modal opens, focus is set depending on whether sensible default choices
+are defined or not.
+
+#### If the user must complete fields before the export can be executed:
+
+1. On opening the modal, the focus is applied to the first field that accepts
+   user input.
+2. On `tab`, progress through any further fields that accept user input.
```
Comment (jillianhowarth) [inline src/pages/community/patterns/export-pattern/index.mdx:179]:
```suggestion
2. On `Tab`, progress through any further fields that accept user input.
```

File: src/pages/community/patterns/export-pattern/index.mdx [hunk 7]
```diff
@@ -150,3 +151,47 @@ A notification can be used to alert the user that an export has been successful.
...
+3. On `tab`, apply focus to "Export".
```
Comment (jillianhowarth) [inline src/pages/community/patterns/export-pattern/index.mdx:180]:
```suggestion
3. On `Tab`, apply focus to Export.
```

File: src/pages/community/patterns/export-pattern/index.mdx [hunk 8]
```diff
@@ -150,3 +151,47 @@ A notification can be used to alert the user that an export has been successful.
...
+4. On `tab`, apply focus to "Cancel".
```
Comment (jillianhowarth) [inline src/pages/community/patterns/export-pattern/index.mdx:181]:
```suggestion
4. On `Tab`, apply focus to Cancel.
```

File: src/pages/community/patterns/export-pattern/index.mdx [hunk 9]
```diff
@@ -150,3 +151,47 @@ A notification can be used to alert the user that an export has been successful.
...
+
+#### If a sensible default value is provided:
+
+1. On opening the modal, set the focus to the "Export" button.
```
Comment (jillianhowarth) [inline src/pages/community/patterns/export-pattern/index.mdx:185]:
```suggestion
1. On opening the modal, set the focus to the Export button.
```

File: src/pages/community/patterns/export-pattern/index.mdx [hunk 10]
```diff
@@ -150,3 +151,47 @@ A notification can be used to alert the user that an export has been successful.
...
+2. On `tab`, apply focus to "Cancel".
```
Comment (jillianhowarth) [inline src/pages/community/patterns/export-pattern/index.mdx:186]:
```suggestion
2. On `Tab`, apply focus to Cancel.
```

File: src/pages/community/patterns/export-pattern/index.mdx [hunk 11]
```diff
@@ -150,3 +151,47 @@ A notification can be used to alert the user that an export has been successful.
...
+3. On `tab`, apply focus to the first field that accepts user input. Follow
```
Comment (jillianhowarth) [inline src/pages/community/patterns/export-pattern/index.mdx:187]:
```suggestion
3. On `Tab`, apply focus to the first field that accepts user input. Follow
```

File: src/pages/community/patterns/export-pattern/index.mdx [hunk 12]
```diff
@@ -150,3 +151,47 @@ A notification can be used to alert the user that an export has been successful.
...
+   accessibility guidance for the interaction type of this field (i.e.
```
Comment (jillianhowarth) [inline src/pages/community/patterns/export-pattern/index.mdx:188]:
```suggestion
   accessibility guidance for the interaction type of this field (for example
```

File: src/pages/community/patterns/export-pattern/index.mdx [hunk 13]
```diff
@@ -150,3 +151,47 @@ A notification can be used to alert the user that an export has been successful.
...
+   [Radio button](https://www.carbondesignsystem.com/components/radio-button/accessibility#how-it-works)
+   or
+   [Text input](https://www.carbondesignsystem.com/components/text-input/accessibility#how-it-works))
+4. On `tab`, progress through any further fields that accept user input.
```
Comment (jillianhowarth) [inline src/pages/community/patterns/export-pattern/index.mdx:192]:
```suggestion
4. On `Tab`, progress through any further fields that accept user input.
```

File: src/pages/community/patterns/export-pattern/index.mdx [hunk 14]
```diff
@@ -150,3 +151,47 @@ A notification can be used to alert the user that an export has been successful.
...
+
+### Feedback
+
+Where passive modals are used to alert the user of success or failure, open the
+modal with focus applied to the "Close" icon.
```
Comment (jillianhowarth) [inline src/pages/community/patterns/export-pattern/index.mdx:197]:
```suggestion
modal with focus applied to the close icon.
```

**Review by jillianhowarth — CHANGES_REQUESTED** [review]
This is awesome @chloepoulterdesign! I have some small content tweaks, but otherwise it's looking great! 🙂

File: src/pages/community/patterns/export-pattern/index.mdx (hunk 14 above)
Comment (jillianhowarth) [reply src/pages/community/patterns/export-pattern/index.mdx:197]: @chloepoulterdesign I'm guessing this sentence wasn't meant to go under the feedback section?

Comment (chloepoulterdesign) [reply src/pages/community/patterns/export-pattern/index.mdx:197]: I think that feedback section is badly named - it's meant to refer to the user interaction where they recieve feedback on the successful export. Will rename!

**Review by jillianhowarth — APPROVED** [review]
Thanks @chloepoulterdesign!

**Review by aledavila — APPROVED** [review]

**Pull request merged**